lib/src/lib.rs

```diff
@@ -348,3 +348,35 @@ impl Display for Rect {
         )
     }
 }
+
+/// A wrapper for the 'v4l2_frmivalenum' union member types
+#[derive(Debug)]
+pub enum FrmIvalTypes {
+    Discrete(bindings::v4l2_fract),
+    StepWise(bindings::v4l2_frmival_stepwise),
+}
+
+impl bindings::v4l2_frmivalenum {
+    pub fn union(&self) -> Option<FrmIvalTypes> {
+        match self.index {
+            // SAFETY: the member of the union that gets used by the driver
+            // is determined by the index
+            bindings::v4l2_frmivaltypes_V4L2_FRMIVAL_TYPE_DISCRETE => {
+                Some(FrmIvalTypes::Discrete(unsafe {
+                    self.__bindgen_anon_1.discrete
+                }))
+            }
+
+            // SAFETY: the member of the union that gets used by the driver
+            // is determined by the index
+            bindings::v4l2_frmivaltypes_V4L2_FRMIVAL_TYPE_CONTINUOUS
+            | bindings::v4l2_frmivaltypes_V4L2_FRMIVAL_TYPE_STEPWISE => {
+                Some(FrmIvalTypes::StepWise(unsafe {
+                    self.__bindgen_anon_1.stepwise
+                }))
+            }
+
+            _ => None,
+        }
+    }
+}
```

Comment:
Was also thinking, how about moving this to `ioctl/frameintervals.rs`? If we start writing implementations on top of V4L2 type to make them safer (which I think is a great idea btw!), this is maybe a more suitable place.